### PR TITLE
Pr/blockchain public address

### PR DESCRIPTION
**Pull request by jordonezlucena**

### CODE REVIEW

**Comment by eric-murray** [comment]
Why is MOM-2023-02-02.md part of this PR? This document is already in the main branch.

On the Blockchain Public Address proposal, is it proposed that this be a new sub project? A first read through suggests it would fit into the new Identity and Consent sub project.

**Comment by jordonezlucena** [comment]
Hi, the inclusion of MoM was a mistake -- I'll fix it later. 
In relation to the new API proposal, the original plan was to work it out in a separate sub-project. The Identity and Consent Sub-Project main focus is on consent, with the possibility of defining new APIs on identity (e.g. Get Identity API) depending on the outcomes of discussions we're having with GSMA. That's why we prefer to keep both tracks separate. Nonetheless, where the new API proposal should be allocated is up to the SteerCo. From TEF, our recommendation is to define a separate sub-project; however, the final decision is on the SteerCo.

**Comment by hdamker** [comment]
Some questions:
- Is my understanding correct, that the proposal is not describing the exposure of a network service, but a kind of (distributed across operators) database service which is based on the ability of the operator to verify subscriber phone numbers?

- Are there further differences between these two files except the filenames? ... maybe one can be deleted from the PR:
  - APIBacklog/documentation/SupportingDocuments/others/BlockchainPublicAddress.yaml
  - APIBacklog/documentation/SupportingDocuments/others/bockchain-public-address.yaml

- What does the abbreviation "OB" in "OB network" standing for?

Beyond these question I suppose there will be several topics to be clarified, but maybe in the (sub)project, e.g.:
- what happens if a number will be ported from one operator to another to which one or more DIDs are bound?
- what happens if a subscriber will lose the access to their device/SIM/subscription? Will this create additional support calls? 
- will operators have financial liabilities if wrong/fraudelent bindings will lead to financial loses?

**Comment by diegogonmar** [comment]
Hi Herbert, 
Sorry for delay in response, here are the answer to your questions:

_Is my understanding correct, that the proposal is not describing the exposure of a network service, but a kind of (distributed across operators) database service which is based on the ability of the operator to verify subscriber phone numbers?_

It’s not a network service as we’re not exposing a network capability here. The proposal is actually to build a new telco service based on a key piece of information in contractual relationship of a user with the telco operator, which is the phone_number (MSISDN).
The new telco service uses phone_number - which is a friendly and universal identifier for the end user - as primary key to associate a (set of) blockchain public address. 
This association will then enable simpler ways to offer several final services by developers using CAMARA APIs. E.g.: send tokens (from my phone_number) to one person in my agenda (to another phone_number)

—

Regarding the two .yaml files, that was a mistake, thanks for pointing it out. My colleague @jordonezlucena will fix it.

—

OB stands for Operating Business, i.e.: telco operator. We use this terminology internally, we can change it for better understanding, also to use same terms in every CAMARA API 

—

As you say, several topics should be discussed in the sub-project, regarding the ones you raise:

_what happens if a number will be ported from one operator to another to which one or more DIDs are bound?_

In portability processes, both the operator porting out the number and the one porting in the number are aware and involved in the process. So, as part of that portability process, one operator can perform the port of the bindings to the other, by using the API (POST) with a priviledged access. 
Obviously, this will not be part of the API exposed to developers, as is a telco 2 telco internal operation.


_what happens if a subscriber will lose the access to their device/SIM/subscription? Will this create additional support calls?_

Well this is yet another service provided to the user. So if subscriber losts device/SIM/subscription there will be support calls requesting service temporary suspenssion of Calls, Texts, browsing, purchases, etc. This will then also imply block this API usage for lost phone_number.

_will operators have financial liabilities if wrong/fraudelent bindings will lead to financial loses?_

The terms and conditions of API usage shall clarify this, and this should be clearly defined by legal departments of the companies. In the same way as other APIs dealing with high sensitive information such as location.

**Comment by jordonezlucena** [comment]
Dear all,

Wrong files in the PR have been removed. We now only have the filled-in template together with the YAML file. 
As per chairman suggestion, the proposal is to close this PR by the end of next week, so we can take it for approval at next SteerCo, which will be held in April. 

@hdamker: hope Diego's points above shed light on your questions. Otherwise, let us know.

**Comment by MarkCornall** [comment]
Dear All,
I am not sure that this belongs into the scope of CAMARA now. My understanding is that CAMARA should define northbound service APIs for developers to use. This proposal is for a new Blockchain system to be operated by operators. Not for a new NB API. That capability definition should fit under other standardisation bodies such as 3GPP or GSMA and I am sure others and not under CAMARA

Mark

**Comment by jordonezlucena** [comment]
> Dear All, I am not sure that this belongs into the scope of CAMARA now. My understanding is that CAMARA should define northbound service APIs for developers to use. This proposal is for a new Blockchain system to be operated by operators. Not for a new NB API. That capability definition should fit under other standardisation bodies such as 3GPP or GSMA and I am sure others and not under CAMARA

The proposed API does represent a northbound service APIs to developer to use. It is NOT an API used among telcos to implement blockchain; instead, it represents a identity mapping solution that can be used to know which is the user identity in the web3 world (i.e. DID) out from user's telephone number. This is pretty useful for developers. 
There are different use cases that this API enables for developers: 
* I want to implement something similar to a bizum for blockchain, since it is easy for users to manage telephone number identifiers, instead DID (6-35 alphanumeric characters in length).
* Upon implementing DAO in a company, I wantsto have the ability able to know the users that I have identified by phone, and who are in the web3 world to interact with them. 
* From any app, e.g. gaming app, I want to send cryptocurrency to a particular user.

**Review by MarkCornall — APPROVED** [review]
All ok

File: APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+| **Field** | Description | 
+| ---- | ----- |
+| API family name | Blockchain Public Address |
+| API family owner| Telefonica  |
+| API summary | **High level description**<br> This API allows to manage a Blockchain Public Address associated to a phone number, i.e. to retrieve the blockchain public address(es) and to bind/unbind a Blockchain Public Address. The Block Public Address is referred to as Decentralized IDentifier (DID). With the proposed API, telco service providers have the opportunity to provide 3rd parties with the following capability: pairing phone number with DID whenever this 3rd party wants to offer its customers a way to make transactions more easily based on the phone number instead of the DID. <br><br>**Rationale**<br>DID is composed of 26-35 alphanumeric characters in length. This identifier is not memorisable or usable by a person, so there is the possibility of positioning the telephone number as the predominant identifier on web3. This new identity protocol will make it possible to have a system similar to the Telcos' bizum in web3 ecosystem.
```
Comment (Kevsy) [inline APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md:5]: “The Block Public Address is referred to as Decentralized IDentifier (DID)” is not quite right, as there are DID methods that not even use blockchain (e.g. did:web and did:dns). Maybe rephrase as 'This Block public address is utilized as  Decentralized Identifier'

Comment (Kevsy) [inline APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md:5]: "DID is composed of 26-35 alphanumeric characters" 

DIDs have no length restriction (as far as I know); W3C ABNF
did                = "did:" method-name ":" method-specific-id
method-name        = 1*method-char
method-char        = %x61-7A / DIGIT
method-specific-id = *( *idchar ":" ) 1*idchar
idchar             = ALPHA / DIGIT / "." / "-" / "_" / pct-encoded
pct-encoded        = "%" HEXDIG HEXDIG

However, a bitcoin address is indeed a 26-35 alphanumeric character identifier (while a wallet address in Ethereum is a 42-character hexadecimal address). Should this say 'wallet address' instead of DID?

File: APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+| Technical viability | For this API to work, the following is needed: <li> Wallets, which will be in charge of generating the DID.</li><li> Storing this DID together with the user's phone number, with prior validation that the targeted user has telephone number. </li><li>Designing this API for it to be interoperable among the different operators</li>The operators will make this API available to 3rd parties through their API marketplaces. The invokation of this API will allow operators to detect possible fraudulent uses. |
```
Comment (Kevsy) [inline APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md:6]: s/invokation/invocation

2)	I do not understand the statement “The invocation of this API will allow operators to detect possible fraudulent uses.” I assume it should say, the operators have to protect the API against fraudulent use. Is there a proposal to define this?

Also how consent of sending AND receiving user will be managed is still to be defined. In general, the proposed mapping may have problems  from a privacy point of view because if I know your wallet address I can see all your transaction done with this address. i.e. The database will be an attractive target of hackers

**Review by Kevsy — CHANGES_REQUESTED** [review]
Please see the inline comments which can help clarify the proposal

File: APIBacklog/documentation/SupportingDocuments/others/BlockchainPublicAddress.yaml [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+     - **Endpoint to remove the association of a phone_number with a Blockchain Public Address**. This operation is intended to be used with a 3-legged Access Token; this is because only the user of the phone_number must be able to perform an unbind. Note: Due to this, the API Server must validate that the `id` included in the request refers to a bind for a phone_number belonging to the sub in the Access Token and, when identifier_bound_scopes are used, validate that the `id` included in the request refers to a bind for a phone_number appearing in the list of identifier_bound_scopes for the scope protecting this endpoint. As a consequence, this operation will only work with phone_numbers of the OB exposing the API.
```
[the remark this answers is not in the document]
Comment (diegogonmar) [reply APIBacklog/documentation/SupportingDocuments/others/BlockchainPublicAddress.yaml:23]: The lifecycle will be managed accordingly. 
- Upon MSISDN termination, owning operator will remove entries in the DDBB for the MSISDN.
- In termination due to portability, the origin operator will notify destination operator of the existing bindings.
- In MSISDN change, the operator should update the binding from old MSISDN to new MSISDN.

File: APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md (hunk 1 above)
Comment (diegogonmar) [reply APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md]: Text is updated according to your suggestion in this commit: https://github.com/camaraproject/WorkingGroups/pull/154/commits/6bec4f9f8a0c0a35295992f9c57c2c4ced689ad5

Comment (diegogonmar) [reply APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md]: Text is updated in this commit: https://github.com/camaraproject/WorkingGroups/pull/154/commits/6bec4f9f8a0c0a35295992f9c57c2c4ced689ad5
Please check, we reworded it to make sentence clearer by using Bitcoin length just as an example

File: APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md (hunk 2 above)
Comment (diegogonmar) [reply APIBacklog/documentation/SupportingDocuments/API proposals/APIproposal_BlockchainPublicAddress_Telefonica.md]: Text is updated according to your suggestion in this commit: https://github.com/camaraproject/WorkingGroups/pull/154/commits/6bec4f9f8a0c0a35295992f9c57c2c4ced689ad5

Regarding consent, yes, it's very important in this API. There is a sub-project for Identity&Consent that should handle these topics and this API has to adhere to conclusions there.

**Review by MarkCornall — DISMISSED** [review]

**Review by hdamker — DISMISSED** [review]
Proposal to setup the sup project LGTM. I assume that the actual API will be discussed within the sub project then.

**Review by diegogonmar — APPROVED** [review]
LGTM

**Review by diegogonmar — APPROVED** [review]
LGTM

**Review by diegogonmar — APPROVED** [review]

**Review by hdamker — APPROVED** [review]
/LGTM
